platforms/jvm/plugins-java/src/main/java/org/gradle/jvm/component/internal/DefaultJvmSoftwareComponent.java

```diff
@@ -186,4 +192,69 @@ public void withSourcesJar() {
     public JvmFeatureInternal getMainFeature() {
         return mainFeature;
     }
+
+    @Override
+    public void consistentResolution(Action<? super JavaResolutionConsistency> action, Project project) {
+        action.execute(new DefaultJavaResolutionConsistency(project, project.getExtensions().getByType(SourceSetContainer.class), project.getConfigurations()));
+    }
+
+    private static class DefaultJavaResolutionConsistency implements JavaResolutionConsistency {
+        private final Configuration mainCompileClasspath;
+        private final Configuration mainRuntimeClasspath;
+        private final Configuration testCompileClasspath;
+        private final Configuration testRuntimeClasspath;
+        private final SourceSetContainer sourceSets;
+        private final ConfigurationContainer configurations;
+
+        @Inject
+        public DefaultJavaResolutionConsistency(Project project, SourceSetContainer sourceSets, ConfigurationContainer configurations) {
+            this.sourceSets = sourceSets;
+            this.configurations = configurations;
+
+            if (project.getPlugins().hasPlugin(JavaPlugin.class)) {
+                JvmFeatureInternal mainFeature = JavaPluginHelper.getJavaComponent(project).getMainFeature();
+                JvmTestSuite defaultTestSuite = JavaPluginHelper.getDefaultTestSuite(project);
```

Comment:
Similar to removing the `JavaPluginHelper.getJavaComponent` call, we also should remove this call, as the default test suite conceptually belongs to this component. 

If we have a reference to the component, we should have a reference to the component's default test suite. 

This likely means we need to move [this code](https://github.com/gradle/gradle/blob/8276ecd71ab279bea9fd14ddc8037b2e054aa071/platforms/jvm/plugins-java/src/main/java/org/gradle/api/plugins/JavaPlugin.java#L290-L319) into the component and save-off a reference to the suite in this class.